pandas/io/parsers/arrow_parser_wrapper.py

```diff
@@ -163,5 +163,8 @@ def read(self) -> DataFrame:
         elif self.kwds["dtype_backend"] == "numpy_nullable":
             frame = table.to_pandas(types_mapper=_arrow_dtype_mapping().get)
         else:
-            frame = table.to_pandas()
+            if self.kwds.get("dtype") is not None and issubclass(type(self.kwds.get("dtype")), dict):
+                frame = table.to_pandas(types_mapper=self.kwds["dtype"].get)
```

Comment:
You could create a defaultdict when dtype is a scalar to cover this case as well.

But please add tests to check that this works